File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0: return 0;
        if n <= 2: return 1;
        ppprev, pprev, prev = 0, 1, 1;
        for i in range(3, n + 1):
            current = ppprev + pprev + prev;
            ppprev, pprev, prev = pprev, prev, current;
        return prev;

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        if (n == 0 ) return 0;
        if (n == 1 or n == 2) return 1;
        int ppprev = 0, pprev = 1, prev = 1;
        for (int i = 3; i <= n; ++i) {
            int current = prev + pprev + ppprev;
            ppprev = pprev;
            pprev = prev;
            prev = current;
        }
        return prev;
    }
};